test/core/security/credentials_test.cc

```diff
@@ -2177,6 +2180,45 @@ void validate_external_account_creds_service_account_impersonation_request(
                     "Bearer token_exchange_access_token") == 0);
 }
 
+void validate_external_account_creds_serv_acc_imp_custom_lifetime_request(
+    const grpc_http_request* request, const char* host, const char* path,
+    const char* body, size_t body_size, bool /*expect_actor_token*/) {
+  // Check that the body is constructed properly.
+  GPR_ASSERT(body != nullptr);
+  GPR_ASSERT(body_size != 0);
+  GPR_ASSERT(strcmp(body, "scope=scope_1%20scope_2&lifetime=1800s") == 0);
```

Comment:
Is there not a way to pass the expected lifetime? Having the validate function only work with the hardcoded 1800s is a little confusing to me.

Reply:
Unfortunately no, as the method is being called implicitly.